slsim/Sources/QuasarCatalog/number_density_quasar.py

```diff
@@ -0,0 +1,246 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.interpolate import interp1d
+from astropy.table import Table
+from matplotlib.lines import Line2D
+
+""" 
+References: 
+Richards et al. 2005: DOI: 10.1111/j.1365-2966.2005.09096.x 
+Richards et al. 2006: DOI: 10.1086/503559 
+Oguri & Marshall (2010): DOI: 10.1111/j.1365-2966.2010.16639.x.
+"""
+
+
+def M_star(z_value, h=0.72, zeta=2.98, xi=4.05, z_star=1.60):
+    """Calculates the break absolute magnitude of quasars for a given redshift according
+    to Eq (11) in Oguri & Marshall (2010)
+
+    Parameters:
+    :param z_value: Redshift value.
+    :type z_value: float
+    :param h: Hubble constant parameter H0/100, where H0 = 72 km s^-1 Mpc^-1.
+    :param zeta: (1) Best fit value of the observed evolution of the quasar luminosity function from SDSS DR3 survery (Richards et al. 2006)
+    :param xi: (2) Best fit value of the observed evolution of the quasar luminosity function from SDSS DR3 survery (Richards et al. 2006)
+    :param z_star: (3) Best fit value of the observed evolution of the quasar luminosity function from SDSS DR3 survery (Richards et al. 2006)
+    :Note: You can adjust these values if needed
+
+    Returns:
+    :rtype: float
+    :return: M_star value.
+    """
+
+    denominator = (np.sqrt(np.exp(xi * z_value)) + np.sqrt(np.exp(xi * z_star))) ** 2
+    if denominator == 0:
+        return np.nan  # or any other suitable value to indicate an error
+    else:
+        return (
+            -20.90
+            + (5 * np.log10(h))
+            - (
+                2.5
+                * np.log10(
+                    np.exp(zeta * z_value) * (1 + np.exp(xi * z_star)) / denominator
+                )
+            )
+        )
+
+
+def dPhi_dM(M, z_value, alpha=-3.31, beta=-1.45, phi_star=5.34e-6 * (0.72**3)):
```

Comment:
is it possible to have this function being compatible with numpy array inputs for M and z_value?